lib/services/hook.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+var http =  require('http');
+var https =  require('https');
+var url = require('url');
+
+module.exports = hookServiceFactory;
+
+hookServiceFactory.$provide = 'Services.Hook';
+hookServiceFactory.$inject = [
+    'Services.Messenger',
+    '_',
+    'Assert',
+    'Services.Waterline'
+];
+
+function hookServiceFactory(
+    messenger,
+    _,
+    assert,
+    waterline
+) {
+    function HookService(){
+    }
+    
+    /**
+     * Publishes the JSON serialized from 'data' into AMQP based on exchange and routingKey.
+     *
+     * @param {string} name
+     * @param {string} routingKey
+     * @param {Object} data
+     * @param {Object} options
+     * @returns {Promise}
+     */
+    HookService.prototype.publish = function(name, routingKey, data, options) {
+        assert.string(name, 'name');
```

Comment:
These asserts should be enclosed within a `Promise.try` block to avoid throwing a synchronous error.